### PR TITLE
eza: init

### PR DESCRIPTION
**Pull request by cafkafk**
This copies the exa configuration for eza, the official fork of the
program.

Signed-off-by: Christina Sørensen <christina@cafkafk.com>

Refs: https://github.com/eza-community/eza/issues/207, https://github.com/ogham/exa/issues/1243

### Checklist

- [ ] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by cafkafk** [comment]
I've never contributed code to home manager before, so I'm sorry if there are any obvious mistakes.

File: modules/programs/eza.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta.maintainers = [ maintainers.cafkafk ];
```
Comment (ryanccn) [inline modules/programs/eza.nix:6]: I don't think `maintainers` is in scope here (also doesn't seem like you added yourself to `modules/lib/maintainers.nix`

**Review by ryanccn — CHANGES_REQUESTED** [review]

**Comment by ryanccn** [comment]
Also I believe you have to add the module to `modules/modules.nix` for it to be included

**Comment by cafkafk** [comment]
> Also I believe you have to add the module to `modules/modules.nix` for it to be included

This push addresses this, thanks!

**Comment by ryanccn** [comment]
I ran into this same issue when copying the exa configuration and replacing `exa` with `eza`, lol

File: modules/programs/eza.nix (hunk 1 above)
Comment (cafkafk) [reply modules/programs/eza.nix]: > I don't think maintainers is in scope here

Sorry, I think I was too fast, changed it to `lib.pkgs.maintainers`. 

> also doesn't seem like you added yourself to modules/lib/maintainers.nix

The docs said you didn't have to add yourself if you were in the nixpkgs maintainers file. If I can't use nixpkgs I'll add myself to the home manager file.

**Comment by cafkafk** [comment]
> I ran into this same issue (`ezample`) when copying the exa configuration and replacing `exa` with `eza`, lol

:woman_facepalming: I thought I had fixed that before pushing.

File: modules/programs/eza.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  options.programs.eza = {
+    enable = mkEnableOption "eza, a modern replacement for {command}`ls`";
+
+    enableAliases = mkEnableOption "recommended eza aliases (ls, ll…)";
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      ezample = [ "--group-directories-first" "--header" ];
```
Comment (ryanccn) [inline modules/programs/eza.nix:16]:
```suggestion
      example = [ "--group-directories-first" "--header" ];
```

**Comment by cafkafk** [comment]
Okay fixed it.

File: modules/programs/eza.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta.maintainers = [ pkgs.lib.maintainers.cafkafk ];
```
Comment (isabelroses) [inline modules/programs/eza.nix:6]: as you have put `with lib;` the expression can be shortened to:
```suggestion
  meta.maintainers = [ pkgs.lib.maintainers.cafkafk ];
  meta.maintainers = [ maintainers.cafkafk ];
```

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Comment by rycee** [comment]
Btw, I notice that the exa package has been [removed from Nixpkgs](https://github.com/NixOS/nixpkgs/pull/253683). We may therefore want to remove the `programs.exa` module as well?

Doing something like https://github.com/nix-community/home-manager/blob/91341cde4143b10ee66e994a53c35d376ad6cdfb/modules/programs/just.nix should be fine. The error message can recommend using `programs.eza` instead. Would any of you be willing to put together a PR for that?

**Comment by cafkafk** [comment]
> Would any of you be willing to put together a PR for that?

I can give it a try. I'll look at it later today probably.

**Pull request merged**